[dvsim] Documentation (#15601)
lowRISC/opentitan: util/dvsim/doc/_index.md

---

> title: "DVSim tool"

Author: @matutem @moidx 

Would you please provide an early feedback on the structure / contents of this documentation?

CC: @domrizz0

Reviewer: @cfrantz fyi

---
title: "DVSim"
---

> DVSim is a purpose-built build and run system written in Python that caters to running a variety of EDA tool flows using a wide variety of commercially available EDA tools.

Reviewer 2: ```suggestion
DVSim is a build and run system written in Python that runs a variety of EDA tool flows using a wide variety of commercially available EDA tools.
```

Author: Fixed.

There are multiple steps involved in running EDA tool flows.

> DVSim effectively encapsulates them all to provide a single, standardized command-line interface to launch them.

Reviewer 2: ```suggestion
DVSim encapsulates them all to provide a single, standardized command-line interface to launch them.
```

Author: Fixed.

While DVSim was written to support OpenTitan, it can be used for any ASIC project.

All EDA tool flows on OpenTitan are launched using the DVSim tool.
The following flows are currently supported:
- Simulations
- Coverage Unreachability Analysis (UNR)
- Formal (formal property verification (FPV), and connectivity)
- Lint (semantic and stylistic)
- Synthesis
- CDC
- RDC

# Installation

Clone the OpenTitan repository by following the [getting started]({{< relref "doc/getting_started" >}}) steps.
The rest of the documentation will assume `$REPO_TOP` as the root of the local OpenTitan repository.
DVSim is located at `$REPO_TOP/util/dvsim/dvsim.py`.

DVSim relies on the following third-party Python libraries:
- **[Hjson](https://hjson.github.io/)**:

>   to parse the DUT configuration data, which is written in Hjson.

Reviewer 2: ```suggestion
  to parse the Hjson DUT configuration data.
```

Author: Fixed.

- **[Enlighten](https://python-enlighten.readthedocs.io/en/stable/)**:

>   to track the progress of the EDA tool flows (especially simulations running in batch mode) on the console in a more natural and readable way.

Reviewer 2: ```suggestion
  to track the progress of the EDA tool flows on the console in a readable way.
```

Author: Fixed.

- **[Mistletoe](https://pypi.org/project/mistletoe/0.-1/)**:

>   to convert testplan descriptions, as well as the EDA tool flow reports, which are constructed in Markdown format, to HTML.

Reviewer 2: ```suggestion
  to convert Markdown format to HTML, used for testplan descriptions and EDA tool flow reports.
```

Author: Fixed.

- **[Premailer](https://pypi.org/project/premailer/)**:
  to inline a block of CSS into the generated HTML report.

- **[Tabulate](https://pypi.org/project/tabulate/)**:
  to pretty-print tabular data when displaying the report on the console.

These dependencies are already listed in `$REPO_TOP/python_requirements.txt`, which can be installed by running:

```console
pip3 install --user -r $REPO_TOP/python-requirements.txt
```

Note that you may have already done this if you followed the getting started steps.

# DVSim design doc

The [design doc]({{< relref "util/dvsim/doc/design_doc.md" >}}) illustrates some of the requirements and motivations that drove the developement of this tool.

# Reference manual

The [reference manual]({{< relref "reference_manual" >}}) provides a detailed technical information about DVSim's internals.
It is suitable for contributors who wish to understand, maintain, and make enhancements to the DVSim codebase.

# User guide

> The [user guide]({{< relref "util/dvsim/doc/user_guide.md" >}}) is aimed at helping new and exising contributors use DVSim effectively.

Reviewer 2: ```suggestion
The [user guide]({{< relref "util/dvsim/doc/user_guide.md" >}}) helps new and existing contributors use DVSim effectively.
```

Author: Fixed.
